no-littering.el

```diff
@@ -77,36 +77,64 @@ This variable has to be set before `no-littering' is loaded.")
   "The directory where packages place their persistent data files.
 This variable has to be set before `no-littering' is loaded.")
 
-(cl-flet ((etc (file) (expand-file-name (convert-standard-filename file)
-                                        no-littering-etc-directory))
-          (var (file) (expand-file-name (convert-standard-filename file)
-                                        no-littering-var-directory)))
-  (with-no-warnings ; many of these variables haven't been defined yet
-
-;;; Built-in packages
-
-    (setq auto-save-list-file-prefix       (var "saves-"))
-    (setq backup-directory-alist           (list (cons "." (var "backups/"))))
-    (setq bookmark-default-file            (var "bookmarks.el"))
-    (setq desktop-path                     (list (var "desktop/")))
-    (setq desktop-base-file-name           (var "default.el"))
-    (setq desktop-base-lock-name           (var "default.lock"))
-    (setq eshell-directory-name            (var "eshell/"))
-    (setq ido-save-directory-list-file     (var "ido.last"))
-    (setq nsm-settings-file                (var "network-security.data"))
-    (setq org-id-locations-file            (var "org/org-id-locations"))
-    (setq org-registry-file                (var "org/org-registry.el"))
-    (setq recentf-save-file                (var "recentf.el"))
-    (setq save-place-file                  (var "saveplace.el"))
-    (setq savehist-file                    (var "savehist.el"))
-    (setq tramp-persistency-file-name      (var "tramp.el"))
-    (setq trash-directory                  (var "trash/"))
-    (setq url-configuration-directory      (var "url/"))
-
-;;; Third-party packages
-
-    (setq smex-save-file                   (var "smex-items"))
-    ))
+(defun no-littering-make-path (base-dir package file)
+  "Returns path BASE-DIR/[PACKAGE/]FILE, creating the necessary
+directories if necessary."
+  (let ((dir (concat base-dir (when package (symbol-name package)))))
+    (unless (file-exists-p dir)
+      (make-directory dir t))
+    (expand-file-name (convert-standard-filename file) dir)))
+
+(defmacro no-littering-setq (package &rest sym-val-pairs)
```

Comment:
Not sure about the name yet. While `setq` can be used to set multiple variables, Stefan recommends against doing it and that the ability to set multiple variables using a single form is only maintained for backward compatibility. Personally I have started to move away from setting multiple variables in one form (except in cases where a single expression is expected and where using multiple `setq`s would require the use of `progn`).

But I can't really think of a better name right now.